### PR TITLE
Zachmu/analyzer scope 3

### PR DESCRIPTION
**Pull request by zachmu**
Partial implementation of subqueries with outer scope resolution. Not every subquery works yet, but I believe that this is strictly additive in terms of capabilities: no query that worked before is broken by this change, and many now pass. I want to get this merged in since it includes a few large interface changes and file moves that could conflict with other work. I will verify that this increases pass rates in sqllogictest before merging.

### CODE REVIEW

File: sql/errors.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Liquidata, Inc.
```
Comment (Hydrocharged) [inline sql/errors.go:1]: I like this, we had too much in `core.go`

File: sql/row.go [hunk 2]
```diff
@@ -19,6 +19,18 @@ func (r Row) Copy() Row {
...
+// Append appends all the values in r2 to this row and returns the result
+func (r Row) Append(r2 Row) Row {
+	row := make(Row, len(r)+len(r2))
+	for i := range r {
+		row[i] = r[i]
+	}
+	for i := range r2 {
+		row[i+len(r)] = r2[i]
+	}
+	return NewRow(row...)
```
Comment (Hydrocharged) [inline sql/row.go:31]: Why call `NewRow` rather than just returning the `row`?

Comment (zachmu) [reply sql/row.go:31]: Good call

File: sql/errors.go (hunk 1 above)
Comment (zachmu) [reply sql/errors.go:1]: :D

**Comment by zachmu** [comment]
Going to wait to merge this until I can do regression analysis against dolt

**Pull request merged**